### DISCUSSION TITLE
How to fail a scenario?

### DISCUSSION THREAD

**Discussion by Zodiase** [artilleryio/artillery#1545] [Artillery]
In my case I use processor hooks to prepare things not easily replicated with scenarios and flows. Things like setting up accounts, signing in and preparing user data, all of which are required but not part of the stress test.

When performing the preparations sometimes the requests would fail and I'm wondering if there is a way to bail the entire scenario then and there from inside one of the hooks.

The preparation involves many async requests so pretty much all hooks are written with Promises and currently if anything throws, it seems like that VU process would just hang there until it times out I guess. 

I could catch the exceptions and call `next` anyway, and in my experience somewhere down the line some other error would eventually fail the scenario somehow but I don't think that's a reasonable behavior since depending on the exact logic, unintended side effects might happen. I'd rather terminate the scenario immediately after an exception.

**Comment by hassy** [comment]
hi @Zodiase 👋 

Calling `next()` with an error will fail the scenario. The `.message` property on the `Error` object will be shown in the aggregate metric report.

We're hoping to have proper `async`/`await` support for hooks soon.

**Reply by Zodiase** [reply]
OK that's wonderful! Is this documented somewhere? I didn't find it and was almost going to read the source code.

**Reply by hassy** [reply]
Turns out we didn't document it properly - added notes to the HTTP reference now (https://docs.art/http-reference). Thank you!
